tests/rptest/services/redpanda_cloud.py

```diff
@@ -235,161 +305,192 @@ def _get_region_id(self, cluster_type, provider, region):
+                "name": self.current.name,
+                "productId": self.current.product_id,
                 "spec": {
-                    "clusterType": cluster_type,
+                    "clusterType": self.config.type,
                     "connectors": {
                         "enabled": True
                     },
-                    "installPackVersion": install_pack_ver,
+                    "installPackVersion": self.current.install_pack_ver,
                     "isMultiAz": False,
                     "networkId": "",
-                    "provider": provider,
-                    "region": region,
-                    "zones": zones,
+                    "provider": self.config.provider,
+                    "region": self.config.region,
+                    "zones": self.current.zones,
                 }
             },
-            "connectionType": connection_type,
-            "namespaceUuid": namespace_uuid,
+            "connectionType": self.current.connection_type,
+            "namespaceUuid": self.current.namespace_uuid,
             "network": {
-                "displayName": f"{connection_type}-network-{name}",
+                "displayName":
+                f"{self.current.connection_type}-network-{self.current.name}",
                 "spec": {
-                    "cidr": "10.1.0.0/16",
-                    "deploymentType": cluster_type,
-                    "installPackVersion": install_pack_ver,
-                    "provider": provider,
-                    "regionId": region_id,
+                    "cidr": _cidr,
+                    "deploymentType": self.config.type,
+                    "installPackVersion": self.current.install_pack_ver,
+                    "provider": self.config.provider,
+                    "regionId": self.current.region_id,
                 }
             },
         }
 
-        self._logger.debug(f'body: {json.dumps(body)}')
+    def _get_cluster(self, _id):
+        """
+        Calls CloudV2 API to get cluster info
+        """
+        _endpoint = f"/api/v1/clusters/{_id}"
+        return self.cloudv2._http_get(endpoint=_endpoint)
+
+    def _update_live_cluster_info(self):
+        """
+        Update info from existing cluster (BYOC)
+        """
+        # get cluster data
+        _c = self._get_cluster(self.config.id)
+        # Fill in immediate configuration
+        self.current.isAlive = True if _c['status'][
+            'health'] == 'healthy' else False
+        self.current.name = _c['name']
+        self.current.namespace_uuid = _c['namespaceUuid']
+        self.current.install_pack_ver = _c['spec']['installPackVersion']
+        self.current.region = _c['spec']['region']
+        self.current.region_id = self._get_region_id()
+        self.current.zones = _c['spec']['zones']
+        self.current.network_id = _c['spec']['networkId']
+        self.current.network_cidr = _c['spec']['network']['networkCidr']
+        self.current.product_id = _c['productId']
```

Comment:
it looks like the json returned from `GET /api/v1/clusters/{clusterId}` does not have a `productId` field. see [swagger](https://cloud-api.ppd.cloud.redpanda.com/documentation/static/index.html).

Reply:
Fixed